mat64/dense.go

```diff
@@ -65,6 +65,11 @@ type Dense struct {
 	capRows, capCols int
 }
 
+// NewDense creates a new matrix of type *Dense with dimensions r and c. If the mat argument is nil,
+// a new data slice is allocated.
+// Data slice is of Row-major order, i.e., the (i, j) element in matrix should be at (i*c + j)-th
+// position in mat.
+// Also note that NewDense(0, 0, nil) can be used for undetermined size matrix initialization.
```

Comment:
1. Where can I find those examples? Or, where should I put these as examples?
2. I don't think the package-level doc is covering this behavior -- at least it is not that clear to me. For example, I don't even know the usage `NewDense(0, 0, mat)` until you mentioned because there is no any doc about it.